arangod/Aql/OptimizerRulesFeature.cpp

```diff
@@ -267,6 +267,10 @@ void OptimizerRulesFeature::addRules() {
                                       OptimizerRule::applyGeoIndexRule_pass6, DoesNotCreateAdditionalPlans, CanBeDisabled);
 
   if (arangodb::ServerState::instance()->isCoordinator()) {
+
+  registerRule("optimize-cluster-single-document-operations", substituteClusterSingleDocumentOperations,
+                 OptimizerRule::substituteSingleDocumentOperations_pass6, CreatesAdditionalPlans, CanBeDisabled);
```

Comment:
the "CreatesAdditionalPlans" is not true here. The rule will only modify an existing plan in place.